Supervision Work/src/uk/ac/cam/ap886/oopjava/supervisionProjects/Hangman/GUIGamePanel.java

```diff
@@ -0,0 +1,134 @@
+package uk.ac.cam.ap886.oopjava.supervisionProjects.Hangman;
+
+import java.awt.Dimension;
+import java.awt.Graphics;
+import java.awt.event.KeyEvent;
+import java.awt.event.KeyListener;
+
+import javax.swing.JPanel;
+
+@SuppressWarnings("serial")
+public class GUIGamePanel extends JPanel {
+	private String pressedKey;
+	private int hangmanStage;
+
+	public GUIGamePanel(){
+		super();
+		KeyListener listener = new KeyListener() {
```

Comment:
This is quite long. Consider not using an anonymous class, but pulling this out into its own file.